domain/src/test/java/org/oppia/domain/audio/AudioPlayerControllerTest.kt

```diff
@@ -0,0 +1,252 @@
+    assertThat(audioPlayerController.isPlaying()).isFalse()
+  }
+
+  @Test
+  fun testAudioPlayer_play_isPlaying() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.play()
+
+    assertThat(audioPlayerController.isPlaying()).isTrue()
+    assertThat(shadowMediaPlayer.isReallyPlaying).isTrue()
+  }
+
+  @Test
+  fun testAudioPlayer_pause_notIsPlaying() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.pause()
+
+    assertThat(audioPlayerController.isPlaying()).isFalse()
+    assertThat(shadowMediaPlayer.isReallyPlaying).isFalse()
+  }
+
+  @Test
+  fun testAudioPlayer_seekTo_hasCorrectProgress() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.seekTo(500)
+
+    assertThat(shadowMediaPlayer.currentPositionRaw).isEqualTo(500)
+  }
+
+  @Test
+  fun testAudioPlayer_releaseMediaPlayer_isReleasedState() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.releaseMediaPlayer()
+
+    assertThat(shadowMediaPlayer.state).isEqualTo(ShadowMediaPlayer.State.END)
+  }
+
+  @Test
+  fun testAudioObserver_invokePrepare_capturesPreparedState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+
+    arrangeMediaPlayer()
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isSuccess()).isTrue()
+    assertThat(audioPlayerResultCaptor.value.getOrThrow().type).isEqualTo(AudioPlayerController.PlayStatus.PREPARED)
+  }
+
+  @Test
+  fun testAudioObserver_invokeCompletion_capturesCompletedState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+    arrangeMediaPlayer()
+
+    shadowMediaPlayer.invokeCompletionListener()
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isSuccess()).isTrue()
+    assertThat(audioPlayerResultCaptor.value.getOrThrow().type).isEqualTo(AudioPlayerController.PlayStatus.COMPLETED)
+  }
+
+  @Test
+  fun testAudioObserver_invokeChangeDataSource_capturesPendingState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+    arrangeMediaPlayer()
+
+    audioPlayerController.changeDataSource(TEST_URL)
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isPending()).isTrue()
+  }
+
+  @Test
+  fun testAudioObserver_invokePlay_capturesPlayingState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+    arrangeMediaPlayer()
+
+    audioPlayerController.play()
+    audioPlayerController.pause()
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isSuccess()).isTrue()
+    assertThat(audioPlayerResultCaptor.value.getOrThrow().type).isEqualTo(AudioPlayerController.PlayStatus.PAUSED)
```

Comment:
Are there any tests verifying that we get a PLAYING state from the LiveData?

Reply:
Added tests